code/game/objects/machinery/squad_supply/supply_beacon.dm

```diff
@@ -173,3 +158,23 @@
 		"You deactivate [src]")
 		H.put_in_active_hand(src)
 
+/datum/supply_beacon
+	var/name = ""
+	var/turf/location
+	var/registered = FALSE
+
+/datum/supply_beacon/proc/register(_name, turf/_location, life_time = 0 SECONDS)
+	if(registered)
+		return
+	name = _name
+	location = _location
+	GLOB.supply_beacon_list |= src
+	registered = TRUE
+	if(!life_time)
+		return
+	addtimer(CALLBACK(src, .proc/unregister), life_time)
+
+/datum/supply_beacon/proc/unregister()
+	GLOB.supply_beacon_list -= src
+	SEND_SIGNAL(src, COMSIG_BEACON_UNREGISTERED)
+	registered = FALSE
```

Comment:
why is this needed in here and a datum ? make a component

Reply:
why would this need to be a component?

Reply:
because you already made it like one ?

Reply:
It would use to much getcomponent to be a proper component